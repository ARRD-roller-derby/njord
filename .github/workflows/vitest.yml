name: Vitest
on: push
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-node@v2
      with:
        node-version: '16.x'
    - name: Install dependencies
      run: npm ci
    - name: Run Vitest tests
      run: npm run coverage
    - uses: actions/upload-artifact@v2
      if: always()
      with:
        name: vitest-report
        path: vitest-report/
        retention-days: 30