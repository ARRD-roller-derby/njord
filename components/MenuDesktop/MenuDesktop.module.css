.container {
  padding: 10px;
  height: 100%;
  transition: 300ms ease-in-out;
  display: flex;
  flex-direction: column;
  gap: 15px;
  overflow-y: auto;
  background: var(--color-bg);
}

.link {
  display: flex;
  gap: 5px;
  align-items: center;
  font-size: 12px;
  position: relative;
}

.title {
  position: relative;
}

.link[data-current='true'] .title::before {
  content: '';
  position: absolute;
  width: 4px;
  height: 4px;
  border-radius: 50%;
  background: var(--color-tierce);
  bottom: -6px;
  left: calc(50% - 2px)
}