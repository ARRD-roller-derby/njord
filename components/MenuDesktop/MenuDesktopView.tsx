import Image from 'next/image'
import Link from 'next/link'
import classes from './MenuDesktop.module.css'
import { MenuLinks, MenuLinksInterface } from '../../datasources/MenuLinks'

interface props {
  readonly iconSize: number
  readonly filter: Function
  readonly pathname: string
}

export default function MenuDesktopView({ iconSize, filter, pathname }: props) {
  return (
    <nav className={classes.container}>
      {MenuLinks.filter((o) => filter(o)).map((link: MenuLinksInterface) => (
        <Link href={link.href} key={link.title} passHref>
          <a
            className={classes.link}
            data-current={
              link.href === '/'
                ? pathname === link.href
                : !!pathname.includes(link.href)
            }
          >
            <Image
              src={link.icon}
              alt={link.title}
              width={iconSize}
              height={iconSize}
            />
            <div>{link.title}</div>
          </a>
        </Link>
      ))}
    </nav>
  )
}
