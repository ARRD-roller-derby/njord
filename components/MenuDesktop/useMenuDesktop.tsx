import { useSession } from "next-auth/react"
import { useRouter } from "next/router"
import { MenuLinksInterface } from "../../datasources/MenuLinks"

export default function useMenuDesktop(){
  const { data: session } = useSession(),
  {pathname} = useRouter(),
  iconSize = 15

  function filter(o:MenuLinksInterface){
    return o.admin ? session?.isAdmin : true
  }

  return {iconSize,filter,pathname}
}