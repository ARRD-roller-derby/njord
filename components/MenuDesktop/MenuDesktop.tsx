import useMenuDesktop from './useMenuDesktop'
import MenuDesktopView from './MenuDesktopView';

export default function MenuDesktop() {
  const props= useMenuDesktop();

  return <MenuDesktopView {...props }/>
}
