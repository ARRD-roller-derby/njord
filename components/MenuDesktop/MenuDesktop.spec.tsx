import { afterEach, describe, expect, it } from 'vitest'
import { render, cleanup } from '@testing-library/react'
import MenuDesktop from './MenuDesktop'

describe('<Menu />', () => {
  afterEach(cleanup)

  it('Check snapshot', () => {
    const { asFragment } = render(<MenuDesktop />)
    expect(asFragment()).toMatchSnapshot()
  })
})
