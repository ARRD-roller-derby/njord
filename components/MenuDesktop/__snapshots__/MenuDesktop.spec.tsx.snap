// Vitest Snapshot v1

exports[`<Menu /> > Check snapshot 1`] = `
<DocumentFragment>
  <nav
    class="_container_vaf4v_1"
  >
    <a
      class="_link_vaf4v_12"
      data-current="true"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2715%27%20height=%2715%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Accueil"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <div>
        Accueil
      </div>
    </a>
    <a
      class="_link_vaf4v_12"
      data-current="false"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2715%27%20height=%2715%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Calendrier"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <div>
        Calendrier
      </div>
    </a>
    <a
      class="_link_vaf4v_12"
      data-current="false"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2715%27%20height=%2715%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Ma league"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <div>
        Ma league
      </div>
    </a>
    <a
      class="_link_vaf4v_12"
      data-current="false"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2715%27%20height=%2715%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Mes adresses"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <div>
        Mes adresses
      </div>
    </a>
    <a
      class="_link_vaf4v_12"
      data-current="false"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2715%27%20height=%2715%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Mon inventaire"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <div>
        Mon inventaire
      </div>
    </a>
    <a
      class="_link_vaf4v_12"
      data-current="false"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2715%27%20height=%2715%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="News"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <div>
        News
      </div>
    </a>
  </nav>
</DocumentFragment>
`;
